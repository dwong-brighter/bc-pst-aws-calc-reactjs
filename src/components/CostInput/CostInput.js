import React, { Component } from 'react';
import NumberFormat from 'react-number-format';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  currencyAmountField: {
    margin: theme.spacing.unit,
  }
});

/**
 * Formatter for a text field to become a currency field.
 * 
 * @param {*} props 
 */
function NumberFormatCurrency (props) {
  const { currencyCode, inputRef, onChange, ...other } = props;

  return (
    <NumberFormat
      {...other}
      getInputRef={inputRef}
      onValueChange={values => {
        onChange({
          target: {
            value: values.value,
          },
        });
      }}
      prefix="$"
      thousandSeparator
      decimalScale={2}
      fixedDecimalScale
      suffix={' ' + currencyCode}
    />
  );
}

NumberFormatCurrency.propTypes = {
  currencyCode: PropTypes.string.isRequired,
  inputRef: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

class CostInput extends Component {
  state = {
    mainAmount: '0',
    cadRounding: '0',
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value,
    });
  }

  render() {
    const { mainAmount, cadRounding } = this.state;
    const { classes, name, description, currencyCode } = this.props;

    var cadRoundingName = name + 'CadRounding';
    return (
      <div className={classes.container}>
        <TextField
          className={classes.currencyAmountField}
          label={description}
          value={mainAmount}
          onChange={this.handleChange('mainAmount')}
          id={name}
          InputProps={{
            inputComponent: (props) =>
              <NumberFormatCurrency currencyCode={currencyCode} {...props} />,
          }}
        />
        <TextField
          className={classes.currencyAmountField}
          label="Rounding"
          value={cadRounding}
          onChange={this.handleChange('cadRounding')}
          id={cadRoundingName}
          InputProps={{
            inputComponent: (props) =>
              <NumberFormatCurrency currencyCode="CAD" {...props} />,
          }}
        />
      </div>
    );
  }
}

CostInput.propTypes = {
  classes: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  currencyCode: PropTypes.string.isRequired,
};

export default withStyles(styles)(CostInput);
