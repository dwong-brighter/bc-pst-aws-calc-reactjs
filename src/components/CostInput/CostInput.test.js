import React from 'react';
import { shallow } from 'enzyme';
import CostInput from './CostInput';

describe('<CostInput />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <CostInput
        name="someCost"
        description="some cost"
        currencyCode="USD"
        initialMainAmount={42}
        initialCadRounding={0.42}
        onMainAmountChange={Function}
        onCadRoundingChange={Function}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
