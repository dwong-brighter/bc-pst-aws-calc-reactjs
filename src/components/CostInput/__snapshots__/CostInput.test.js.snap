// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CostInput /> renders without crashing 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "inputComponent": [Function],
      }
    }
    className="makeStyles-currencyAmountField-2"
    id="someCost"
    label="some cost"
    onChange={[Function]}
    value={42}
  />
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "inputComponent": [Function],
      }
    }
    className="makeStyles-currencyAmountField-2"
    id="someCostCadRounding"
    label="Rounding"
    onChange={[Function]}
    value={0.42}
  />
</Styled(MuiBox)>
`;
